File: apps/pos_app/lib/models/product.dart
class Product {
  final String id;
  final String nombre;
  final String? barcode;
  final double precioVenta;
  final double stockActual;
  final bool favorito;

  Product({
    required this.id,
    required this.nombre,
    required this.barcode,
    required this.precioVenta,
    required this.stockActual,
    this.favorito = false,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as String,
      nombre: json['nombre'] as String,
      barcode: json['barcode'] as String?,
      precioVenta: (json['precioVenta'] as num).toDouble(),
      stockActual: (json['stockActual'] as num).toDouble(),
      favorito: json['favorito'] as bool? ?? false,
    );
  }
}

File: apps/pos_app_legacy/app/src/main/kotlin/com/tienda/pos_legacy/model/Product.kt
package com.tienda.pos_legacy.model

import org.json.JSONObject

data class Product(
    val id: String,
    val nombre: String,
    val barcode: String?,
    val precioVenta: Double,
    val stockActual: Double,
) {
    companion object {
        fun fromJson(json: JSONObject): Product = Product(
            id = json.getString("id"),
            nombre = json.getString("nombre"),
            barcode = if (json.isNull("barcode")) null else json.getString("barcode"),
            precioVenta = json.getDouble("precioVenta"),
            stockActual = json.getDouble("stockActual"),
        )
    }
}
